dot/parachain/backing/statement_table.go

```diff
@@ -29,24 +32,74 @@ type proposal struct { //nolint:unused
 	signature     parachaintypes.Signature
 }
 
-type candidateData struct { //nolint:unused
+type candidateData struct {
 	groupID       parachaintypes.ParaID
 	candidate     parachaintypes.CommittedCandidateReceipt
 	validityVotes map[parachaintypes.ValidatorIndex]validityVoteWithSign
 }
 
-type validityVoteWithSign struct { //nolint:unused
+// attested yields a full attestation for a candidate.
+// If the candidate can be included, it will return attested candidate.
+func (data candidateData) attested(validityThreshold uint) (*AttestedCandidate, error) {
+	numOfValidityVotes := uint(len(data.validityVotes))
+	if numOfValidityVotes < validityThreshold {
+		return nil, fmt.Errorf("%w: %d < %d", errNotEnoughValidityVotes, numOfValidityVotes, validityThreshold)
+	}
+
+	validityAttestations := make([]validityAttestation, 0, numOfValidityVotes)
+	for validatorIndex, voteWithSign := range data.validityVotes {
+		switch voteWithSign.validityVote {
+		case valid:
+			attestation := parachaintypes.NewValidityAttestation()
+			err := attestation.Set(parachaintypes.Explicit(voteWithSign.signature))
+			if err != nil {
+				return nil, fmt.Errorf("failed to set validity attestation: %w", err)
+			}
+
+			validityAttestations = append(validityAttestations, validityAttestation{
+				ValidatorIndex:      validatorIndex,
+				ValidityAttestation: attestation,
+			})
+		case issued:
+			attestation := parachaintypes.NewValidityAttestation()
+			err := attestation.Set(parachaintypes.Implicit(voteWithSign.signature))
+			if err != nil {
+				return nil, fmt.Errorf("failed to set validity attestation: %w", err)
+			}
+
+			validityAttestations = append(validityAttestations, validityAttestation{
+				ValidatorIndex:      validatorIndex,
+				ValidityAttestation: attestation,
+			})
+		default:
+			return nil, fmt.Errorf("unknown validity vote: %d", voteWithSign.validityVote)
+		}
+	}
+
+	sort.Slice(validityAttestations, func(i, j int) bool {
+		return validityAttestations[i].ValidatorIndex < validityAttestations[j].ValidatorIndex
+	})
```

Comment:
I've been using the slices package for sorting slices.  Check out [slices.SortFunc](https://pkg.go.dev/slices#SortFunc)